mifosng-android/src/main/java/com/mifos/mifosxdroid/online/ClientActivity.java

```diff
@@ -146,4 +151,15 @@ public void loadSurveyQuestion(Survey survey, int clientId) {
         myIntent.putExtra(Constants.CLIENT_ID, clientId);
         startActivity(myIntent);
     }
+
+    @Override
+    public void updateUI() {
```

Comment:
Reattaching the fragment would lead to network calls again. Can you check if update can be accomplished by something similar: https://github.com/openMF/android-client/pull/767 

As you already have the name of the file, you only need the url to download it which I believe, should be received on successful uploading. 

Let me know if such a thing is not possible because of the response you are receiving.